whatshap/cli/haplotag.py

```diff
@@ -378,8 +381,11 @@ def open_output_alignment_file(aln_output, reference, vcf_md5, bam_header):
             )
         kwargs = dict(mode="wc", reference_filename=reference)
     else:
-        # Write BAM
-        kwargs = dict(mode="wb")
+        # Write BAM, disable compression for stdout
+        if aln_output is sys.stdout:
+            kwargs = dict(mode="wb0", threads=threads)
```

Comment:
Out of curiosity: How did you find which mode string to use? Mode `wb0` is listed in the pysam documentation, but it’s not documented what it does.

Reply:
From knowledge of what htslib does with these modes (look through the samtools view source for example). The number is the compression level.